### DISCUSSION TITLE
updateQuery not working

### DISCUSSION THREAD

**Discussion by bubbaspaarx** [urql-graphql/urql#2604] [Q&A] [answered]
Hi there,

We are building a drag and drop area which requires and initial call for some a list of lists which each contain an 'rosterOrder' and 'channels'. 
With the drag and drop, we are making mutations for the rosterOrder and the channels separately and this is causing some issues with our drag and drop flickering. My initial thought was to manually update the cache to reduce the roundtrips and hopefully prevent the issues we are seeing. 

I have the initial query 

```
export const CampaignListsQuery = `
query ($campaignId:ID!) {
  campaignLists(campaignId: $campaignId) {
    id
    rosterOrder
    channels{
      id
      approved
      declined
      pending
      }
    }
```

When we run the updateRosterOrder mutation

```
export const UpdateRosterOrderMutation = `
mutation ($campaignListId: ID!, $rosterOrder: [Int!]!) {
  updateRosterOrder(campaignListId: $campaignListId, rosterOrder: $rosterOrder) {
    id
    rosterOrder
  }
}
`
```

I am trying to use the cache.updateQuery method to update just the rosterOrder of the campaignList I am seeking. 

The issue is that I am receiving an error
```
Uncaught TypeError: Cannot read properties of undefined (reading 'campaignLists')
```

updateQuery

```

      updateRosterOrder: (result, args, cache, _info) => {
        const CampaignListQuery = gql`
          query ($campaignId: ID!) {
            campaignLists(campaignId: $campaignId) {
              id
              rosterOrder
              campaignListChannels {
                id
                approved
                declined
                pending
              }
            }
          }
        `
        cache.updateQuery({ query: CampaignListQuery, variables: { campaignId: '477' } }, (data) =>
          console.log(data)
        )
```
I hard coded the campaignId as I wasn't initially sending that through but I can change that going forward.

**Comment by JoviDeCroock** [comment, marked as answer]
So when are you getting the error? If you do `return data;` it shouldn't have that issue right? 😅  mainly trying to triangulate where your error is coming from

**Reply by bubbaspaarx** [reply]
Hi Jovi, 

It's happening as soon as I run 
```
    cache.updateQuery({ query: CampaignListQuery, variables: { campaignId: '477' } }, (data) =>
          console.log(data)
        )
```

**Reply by JoviDeCroock** [reply]
Yes but that can be happening due to the callback 😅 you're not returning anything. If that is not the cause we'll need a way to see this ourselves so we can track it or more information
